src/cascadia/TerminalSettingsEditor/ColorSchemesPageViewModel.cpp

```diff
@@ -205,6 +192,24 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         }
     }
 
+    void ColorSchemesPageViewModel::RequestDuplicateCurrentScheme()
+    {
+        if (_CurrentScheme)
+        {
+            if (auto actualCurrentScheme = _viewModelToSchemeMap.TryLookup(_CurrentScheme))
+            {
+                auto scheme = _settings.GlobalSettings().DuplicateColorScheme(actualCurrentScheme);
+                // Construct the new color scheme VM
+                const auto schemeVM{ winrt::make<ColorSchemeViewModel>(scheme, *this, _settings) };
+                _AllColorSchemes.Append(schemeVM);
+                _viewModelToSchemeMap.Insert(schemeVM, scheme);
+                CurrentScheme(schemeVM);
+                CurrentPage(ColorSchemesSubPage::Base);

```

Comment:
We're switching to the base page even though the next line immediately switches to the edit page, I believe this is because of the `Breadcrumb` logic in `MainPage`? Unrelated to this PR I should probably do something about that, this is ugh.

Reply:
YEP that's exactly why. It can't update the page inline.